File: app/db/models/bottle.py
from sqlalchemy import Column, Integer, String, ForeignKey, Text
from sqlalchemy.orm import relationship
from app.db.base import Base

class Bottle(Base):
    __tablename__ = "bottles"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False, index=True)
    brand = Column(String, nullable=True)
    flavor_profile = Column(String, nullable=True)
    capacity_ml = Column(Integer, nullable=True)
    
    # Image stored as base64 data URL
    image_url = Column(Text, nullable=True)
    # Barcode number
    barcode = Column(String, nullable=True, index=True)

    spirit_type_id = Column(Integer, ForeignKey("spirit_types.id"), nullable=True)
    spirit_type = relationship("SpiritType")

    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    user = relationship("User", back_populates="bottles")

File: bottles-app/src/types/Bottle.ts
export interface Bottle {
    id: number;
    name: string;
    brand: string | null;
    flavor_profile: string | null;
    spirit_type_id: number;
    spirit_type?: { id: number; name: string };
    capacity_ml: number;
    image_url: string | null;
    barcode: string | null;
  }
